packages/vscode-extension/src/project/project.ts

```diff
@@ -93,29 +99,48 @@ export class Project implements Disposable, ProjectInterface, DeviceSessionsMana
         this.eventEmitter.emit("licenseActivationChanged", hasActiveLicense);
       })
     );
-
     this.disposables.push(
-      workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
-        if (event.affectsConfiguration("launch")) {
-          const config = getLaunchConfiguration();
-          const oldAppRoot = this.appRootFolder;
-          if (config.appRoot === oldAppRoot) {
-            return;
-          }
-          this.setupAppRoot();
-
-          if (this.appRootFolder === undefined) {
-            window.showErrorMessage(
-              "Unable to find the new app root, after a change in launch configuration. Radon IDE might not work properly.",
-              "Dismiss"
-            );
-            return;
-          }
+      this.launchConfigsManager.onLaunchConfigurationsChanged((launchConfigs) => {
+        this.updateProjectState({
+          customLaunchConfigurations: launchConfigs,
+        });
+        const selectedLaunchConfig =
+          launchConfigs[0] ?? launchConfigurationFromOptions({ appRoot: this.relativeAppRootPath });
+
+        const oldAppRoot = this.applicationContext.appRootFolder;
+        if (selectedLaunchConfig.absoluteAppRoot !== oldAppRoot) {
+          // If the app root has changed, we need to update the application context
+          this.setLaunchConfiguration(selectedLaunchConfig);
+        } else {
+          this.applicationContext.updateLaunchConfig(selectedLaunchConfig);
+          this.updateProjectState({
+            selectedLaunchConfiguration: selectedLaunchConfig,
+          });
         }
       })
     );
   }
 
+  async setLaunchConfiguration(options: LaunchConfigurationOptions): Promise<void> {
```

Comment:
similar comment re the naming here, we take options as input, so using the setter notation doesn't feel right